### PR TITLE
Introduce System Controller v2.0

### PR DESCRIPTION
**Pull request by kammce**
Supports #1136

### CODE REVIEW

File: library/L1_Peripheral/lpc40xx/system_controller.hpp [hunk 1]
```diff
@@ -197,229 +195,359 @@ class SystemController final : public sjsu::SystemController
...
-                        units::frequency::megahertz_t desired_frequency) const
-  {
-    SJ2_ASSERT_FATAL(desired_frequency < 384_MHz && desired_frequency > 12_MHz,
-                     "Frequency must be lower than 384 MHz"
-                     "and greater than or equal to 12 MHz");
-    bool calculating = true;
-    uint32_t multiplier_value;
-    if ((desired_frequency.to<uint32_t>() % input_frequency.to<uint32_t>()) > 0)
+    // =========================================================================
+    // Step 3. Configure PLLs
```
Comment (HuangDave) [inline library/L1_Peripheral/lpc40xx/system_controller.hpp:383]: Should be Step 4. here

File: demos/sjtwo/system_clock/source/main.cpp [hunk 2]
```diff
@@ -2,33 +2,89 @@
...
+      sjsu::lpc40xx::SystemController::ClockConfiguration>();
+  // Select system clock (use crystal oscillator directly)
+  config.cpu.select =
+      sjsu::lpc40xx::SystemController::CpuClockSelect::kSystemClock;
+  // Make sure to utilize the internal oscillator only.
+  config.system_oscillator =
+      sjsu::lpc40xx::SystemController::OscillatorSource::kIrc;
+  // Make sure PLL0 is enabled and available
+  config.pll[0].enabled  = true;
+  config.pll[0].multiply = 4;
```
Comment (HuangDave) [inline demos/sjtwo/system_clock/source/main.cpp:24]: Might be a good note that the main PLL is configured to output 48 MHz.

Comment (kammce) [reply demos/sjtwo/system_clock/source/main.cpp:24]: Fill do.

File: library/L1_Peripheral/lpc40xx/system_controller.hpp (hunk 1 above)
Comment (kammce) [reply library/L1_Peripheral/lpc40xx/system_controller.hpp:403]: Will change.

**Pull request merged**